src/python/qepsi4/_psi4.py

```diff
@@ -231,3 +251,164 @@ def get_ham_from_psi4(
+            saved Hamiltonian. If None, includes all orbitals, else you must provide
+            active orbitals in orbs.
+        n_occupied_extract (int): number of occupied molecular orbitals to
+            include in the saved Hamiltonian. Must be less than or equal to
+            n_active_extract. If None, all occupied orbitals are included,
+            except the core orbitals if freeze_core_extract is set to True.
+        freeze_core_extract (bool): If True, frozen core orbitals will always be
+            doubly occupied in the saved Hamiltonian. Ignored if
+            n_occupied_extract is not None.
+    Returns:
+        rdm (openfermion.ops.InteractionRDM): an openfermion object storing
+            1- and 2-RDMs. 
+    """
+
+    one_rdm_a = np.array(wfn.get_opdm(
+        0, 0, 'A', True)).reshape(wfn.nmo(), wfn.nmo()) # Note: this works even if freeze_core is True
+    one_rdm_b = np.array(wfn.get_opdm(
+        0, 0, 'B', True)).reshape(wfn.nmo(), wfn.nmo())
+
+    # Get 2-RDM from CI calculation.
+    n_core_extract = 0
+    if freeze_core_extract and n_occupied_extract is None:
+        n_core_extract = wfn.nfrzc() # This logic implies that cores are always the lowest energy orbitals.
+    elif n_occupied_extract is not None:
+        if wfn.nalpha() != wfn.nbeta():
+            raise ValueError(
+                f"Requesting a number of occupied molecular orbitals not supported when number of alpha and beta electrons is unequal."
+            )
+        if n_occupied_extract > wfn.nalpha():
+            raise ValueError(
+                f"Number of occupied molecular orbitals to extract ({n_occupied_extract}) is larger than number of occupied molecular orbitals ({wfn.nalpha()})."
+            )
+
+        n_core_extract = wfn.nalpha() - n_occupied_extract
+
+    if n_active_extract is None:
+        n_active_extract = wfn.nmo() - n_core_extract
+    
+    nfnmo = wfn.nmo() - wfn.frzcpi().sum() - wfn.frzvpi().sum() # I wouldn't work otherwise for 2-RDMs
+
+    two_rdm_aa = np.array(wfn.get_tpdm(
+        'AA', False)).reshape(nfnmo, nfnmo,
+                                nfnmo, nfnmo)
+    two_rdm_ab = np.array(wfn.get_tpdm(
+        'AB', False)).reshape(nfnmo, nfnmo,
+                                nfnmo, nfnmo)
+    two_rdm_bb = np.array(wfn.get_tpdm(
+        'BB', False)).reshape(nfnmo, nfnmo,
+                                nfnmo, nfnmo)
+
+    # Indices of active molecular orbitals
+    active_indices = range(n_core_extract, n_core_extract + n_active_extract)
+
+    # adjusting 1-RDM
+    one_rdm_a = one_rdm_a[np.ix_(active_indices, active_indices)]
+    one_rdm_b = one_rdm_b[np.ix_(active_indices, active_indices)]
+    
+    if freeze_core:
+        active_indices = range(n_active_extract) # Need to double check if this actually works
+    # If one also requested freeze_core_extract => we need to construct and save RHF RDMs for the core
```

Comment:
Fair point, thanks a lot for pointing that out! I realized that we may need to rethink the way we handle active space selection options for the purpose of RDM extraction since e.g. if we run with `freeze_core = False` and `n_occupied_extract < n_el / 2` the RDM won't correspond to the correct number of particles.

Reply:
That's right, I did not even think about that! This problem would be solved if `n_occupied_extract` also implied freezing the corresponding number of orbitals in the CI calculation, but that's not the case yet.